pl_examples/basic_examples/simple_image_classifier.py

```diff
@@ -66,47 +76,19 @@ def test_step(self, batch, batch_idx):
     def configure_optimizers(self):
         return torch.optim.Adam(self.parameters(), lr=self.hparams.learning_rate)
 
-    @staticmethod
-    def add_model_specific_args(parent_parser):
-        parser = parent_parser.add_argument_group("LitClassifier")
-        parser.add_argument('--hidden_dim', type=int, default=128)
-        parser.add_argument('--learning_rate', type=float, default=0.0001)
-        return parent_parser
+
+class MyLightningCLI(LightningCLI):
+
+    def before_parse_arguments(self, parser):
+        parser.set_defaults(seed_everything=1234)
+
+    def after_fit(self):
+        result = self.trainer.test(self.model, datamodule=self.datamodule)
```

Comment:
I think calling `trainer.test` in `cli_main` would look nicer than having to subclass for this example

Reply:
Certainly can be in cli_main. Though being examples it might be good to do different things to illustrate ways of using lightning. For instance all examples have test in cli_main except in one which would be like now in after_fit to show this as a possibility.

Reply:
Well, these examples are supposed to facilitate understanding/converting to Lightning, not necessarily to showcase all of LightningCLI's potential. For this reason, I prefer to run test in `cli_main`.

The fact that we got rid of all `argparse`'s boilerplate is already great, so no point in showing LightningCLI "complex" features here.

We could include a new example (in a separate PR) that focuses entirely on complex LightningCLI usage.

Reply:
Makes sense. Simplified where needed.